src/EnergyPlus/WaterThermalTanks.hh

```diff
@@ -247,6 +247,16 @@ namespace WaterThermalTanks {
                                                                                         "OTHERFUEL2",
                                                                                         "DISTRICTHEATING"};
 
+    enum class TankOperatingMode
+    {
+        Invalid = -1,
+        Heating,  // heating source is on, source will not turn off until setpoint temp is reached
+        Floating, // heating source is off, source will not turn on until cut-in temp is reached
+        Venting,  // tank temp is above maximum temperature and water is venting
+        Cooling,  // cooling source is on, source will not turn off until setpoint temp is reached
+        Num
+    };
```

Comment:
Moved these from `int const` variables living on state to a proper enum.  I don't _think_ we need a matching string view array because these are purely internal operating modes, not connected to input strings.